contracts/token/OtcEscrow.sol

```diff
@@ -0,0 +1,91 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.6.10;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import { SafeMath } from "@openzeppelin/contracts/math/SafeMath.sol";
+
+import { Vesting } from "./Vesting.sol";
+
+/*
+    Simple OTC Escrow contract to transfer index in exchange for specified USDC amount
+*/
+contract OtcEscrow {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    event VestingDeployed(address vesting);
+
+    address public usdc;
+    address public index;
+
+    address public indexGov;
+    address public beneficiary;
+
+    uint256 public vestingStart;
+    uint256 public vestingEnd;
+    uint256 public vestingCliff;
+
+    uint256 public usdcAmount;
+    uint256 public indexAmount;
+
+    constructor(
+        address _beneficiary,
+        address _indexGov,
+        uint256 _vestingStart,
+        uint256 _vestingCliff,
+        uint256 _vestingEnd,
+        uint256 _usdcAmount,
+        uint256 _indexAmount,
+        address _usdcAddress,
+        address _indexAddress
+    ) public {
+        beneficiary = _beneficiary;
+        indexGov =  _indexGov;
+
+        vestingStart = _vestingStart;
+        vestingCliff = _vestingCliff;
+        vestingEnd = _vestingEnd;
+
+        usdcAmount = _usdcAmount;
+        indexAmount = _indexAmount;
+
+        usdc = _usdcAddress;
+        index = _indexAddress;
+    }
+
+    modifier onlyApprovedParties() {
```

Comment:
Not sure this modifier makes sense, never really seems like `indexGov` or `beneficiary` should be doing the same action